bindings/python/FloatingBaseEstimators.cpp

```diff
@@ -0,0 +1,147 @@
+
+    py::class_<Options>(module, "FloatingBaseEstimatorOptions")
+        .def(py::init())
+        .def_readwrite("imu_bias_estimation_enabled", &Options::imuBiasEstimationEnabled)
+        .def_readwrite("static_imu_bias_initialization_enabled", &Options::staticImuBiasInitializationEnabled)
+        .def_readwrite("nr_samples_for_imu_bias_initialization", &Options::nrSamplesForImuBiasInitialization)
+        .def_readwrite("ekf_update_enabled", &Options::ekfUpdateEnabled)
+        .def_readwrite("acceleration_due_to_gravity", &Options::accelerationDueToGravity);
+
+    py::class_<SensorsStdDev>(module, "FloatingBaseEstimatorSensorsStdDev")
+        .def(py::init())
+        .def_readwrite("accelerometer_noise", &SensorsStdDev::accelerometerNoise)
+        .def_readwrite("gyroscope_noise", &SensorsStdDev::gyroscopeNoise)
+        .def_readwrite("accelerometer_bias_noise", &SensorsStdDev::accelerometerBiasNoise)
+        .def_readwrite("gyroscope_bias_noise", &SensorsStdDev::gyroscopeBiasNoise)
+        .def_readwrite("contact_foot_lin_vel_noise", &SensorsStdDev::contactFootLinvelNoise)
+        .def_readwrite("contact_foot_ang_vel_noise", &SensorsStdDev::contactFootAngvelNoise)
+        .def_readwrite("swing_foot_lin_vel_noise", &SensorsStdDev::swingFootLinvelNoise)
+        .def_readwrite("swing_foot_ang_vel_noise", &SensorsStdDev::swingFootAngvelNoise)
+        .def_readwrite("forward_kinematics_noise", &SensorsStdDev::forwardKinematicsNoise)
+        .def_readwrite("encoders_noise", &SensorsStdDev::encodersNoise);
+
+    py::class_<Advanceable<Output>>(module, "FloatingBaseEstimatorOutputAdvanceable");
+
+    py::class_<FloatingBaseEstimator, Advanceable<Output>> floatingBaseEstimator(module, "FloatingBaseEstimator");
+
+    floatingBaseEstimator.def(py::init())
+        .def("set_imu_measurement", &FloatingBaseEstimator::setIMUMeasurement,
+             py::arg("acc_meas"), py::arg("gyro_meas"))
+        .def("set_contact_status", &FloatingBaseEstimator::setContactStatus,
+             py::arg("name"), py::arg("contact_status"), py::arg("switch_time"), py::arg("time_now"))
+        .def("set_kinematics", &FloatingBaseEstimator::setKinematics,
+             py::arg("encoders"), py::arg("encoder_speeds"))
+        .def("advance", &FloatingBaseEstimator::advance)
+        .def("reset_estimator", py::overload_cast<const InternalState&>(&FloatingBaseEstimator::resetEstimator),
+             py::arg("new_state"))
+        .def("reset_estimator", py::overload_cast<const Eigen::Quaterniond&, const Eigen::Vector3d&>(&FloatingBaseEstimator::resetEstimator),
+             py::arg("new_base_orientation"), py::arg("new_base_position"))
+        .def("get", &FloatingBaseEstimator::get)
+        .def("is_valid", &FloatingBaseEstimator::isValid)
+        .def("initialize",
+            [](FloatingBaseEstimator& impl, std::shared_ptr<IParametersHandler> handler, std::shared_ptr<iDynTree::KinDynComputations> kinDyn) -> bool {
```

Comment:
I think these files are not formatted with project's clang-format. This line for instance seems way too long.

@GiulioRomualdi I see this in many files of blf. Do you think that a new specific PR that uniforms all the files of this project (and then enforcing it) could be done? In order to minimize conflicts, maybe we can do it in a moment with no open PRs?

Reply:
Thanks @GiulioRomualdi for taking care of the formatting in https://github.com/dic-iit/bipedal-locomotion-framework/pull/218/commits/0a86455e1f2236547b9aed1b503538a73abfb00f